src/Routing/RequestContext.php

```diff
@@ -58,34 +65,62 @@ public function __construct(
         );
         $this->decorated = $decorated;
         $this->pageSlugConditionChecker = $pageSlugConditionChecker;
-        $this->localeContext = $localeContext;
+        $this->localeProvider = $localeProvider;
+        $this->requestStack = $requestStack;
     }
 
     /**
-     * @param string $slug
+     * @param Request $request
      *
      * @return bool
      */
-    public function checkPageSlug(string $slug): bool
+    public function checkPageSlug(Request $request): bool
     {
-        return $this->pageSlugConditionChecker->isPageSlug($this->prepareSlug($slug));
+        $mainRequest = $this->requestStack->getMainRequest();
+        if ($mainRequest !== $request) {
+            return false;
+        }
+        $locale = $this->getLocale($request->getPathInfo());
+        $slug = $this->getSlug($request->getPathInfo(), $locale);
+
+        return $this->pageSlugConditionChecker->isPageSlug($locale, $slug);
+    }
+
+    /**
+     * Retrieve locale from URL because is not set yet on the request.
+     *
+     * @param string $pathInfo
+     *
+     * @return string
+     */
+    private function getLocale(string $pathInfo): string
+    {
+        $availableLocaleCodes = $this->localeProvider->getAvailableLocalesCodes();
+        $parts = explode('/', trim($pathInfo, '/'));
+        foreach ($parts as $part) {
+            if (\in_array($part, $availableLocaleCodes, true)) {
+                return $part;
+            }
+        }
+
+        return $this->localeProvider->getDefaultLocaleCode();
     }
```

Comment:
This doesn't work if the locale isn't in the path.
What about the locale in the subdomain? and so on?

We need the LocaleContext to get the locale.